### ISSUE TITLE
Pulseaudio-equalizer and pulseaudio qpaeq bug

### ISSUE THREAD

**Issue by cyberalex4life** [aleho/gnome-shell-volume-mixer#42]
When trying to use pulseaudio-equalizer or pulseaudio-qpaeq (fftw equalizer) and when using turbo boost and/or changing volume step from default value (6) mixer does not show.

**Comment by cyberalex4life** [comment]
```
Jan 10 13:22:43 alex-ntbk gnome-shell.desktop[24457]: (gnome-shell:24457): Gjs-WARNING **: JS ERROR: TypeError: port is null
Jan 10 13:22:43 alex-ntbk gnome-shell.desktop[24457]: Mixer<._showVolumeOsd@/home/alex/.local/share/gnome-shell/extensions/shell-volume-mixer@derhofbauer.at/mixer.js:519
Jan 10 13:22:43 alex-ntbk gnome-shell.desktop[24457]: wrapper@resource:///org/gnome/gjs/modules/lang.js:169
Jan 10 13:22:43 alex-ntbk gnome-shell.desktop[24457]: Mixer<.changeStreamVolume@/home/alex/.local/share/gnome-shell/extensions/shell-volume-mixer@derhofbauer.at/mixer.js:164
Jan 10 13:22:43 alex-ntbk gnome-shell.desktop[24457]: wrapper@resource:///org/gnome/gjs/modules/lang.js:169
Jan 10 13:22:43 alex-ntbk gnome-shell.desktop[24457]: Mixer<.increaseMasterVolume@/home/alex/.local/share/gnome-shell/extensions/shell-volume-mixer@derhofbauer.at/mixer.js:116
Jan 10 13:22:43 alex-ntbk gnome-shell.desktop[24457]: wrapper@resource:///org/gnome/gjs/modules/lang.js:169
```

**Comment by cyberalex4life** [comment]
Ok so I've been able to put together a workaround. This goes like this: 
In widget.js: 
'''
...............................
const Utils = Extension.imports.utils;
var ITEM_LABEL=""; //added
...............................
const MasterSlider = new Lang.Class({
    Name: 'MasterSlider',
    Extends: StreamSlider,

    _init: function(control, options) {
        ..................................
        this.parent(control, options);
        ITEM_LABEL=this._label; //added
    },
    ..................................
   _updateLabel: function() {
        this._label.text = this._stream.description;
        ITEM_LABEL = this._label.text; //added
    },
...............................
});
'''
In mixer.js
'''
...............................
const Utils = Extension.imports.utils;
const WIDGET = Extension.imports.widget; //added
...............................

const Mixer = new Lang.Class({
    Name: 'ShellVolumeMixerMixer',
    ..................................
    _showVolumeOsd: function(level, percent) {
    ..................................
    	if (this._defaultSink) {
            let port = this._defaultSink.get_port();
            if (!port) {
	            label.push(WIDGET.ITEM_LABEL);
            }
            else if (port.port != 'analog-output-speaker'
                    && port.port != 'analog-output') {
                label.push(port.human_port);
            }
           
        }
    ..................................
});
'''

**Comment by cyberalex4life** [comment]
My experience with javascript is poor, so you may be able to make it look nicer

**Comment by cyberalex4life** [comment]
Issue closed  because of commit c67836dca88e73bb28ab4f4001dda8f4c9adda55

**Comment by aleho** [comment]
Really hard to understand what you were actually trying to accomplish with the broken formatting. A patch might have been better (or even a pull request).
